### PR TITLE
Reduce Mccabe code complexity to 8

### PR DESCRIPTION
**Pull request by joppevos**
## Description

Created this small PR to reduce the maccabe score to 8 and refactored the code flagged as too complex.

I removed flake8, since we can use ruffs maccabe linter.

The following functions have been addressed
```shell
> pre-commit run flake8 --all-files
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

cosmos/airflow/graph.py:196:1: C901 'build_airflow_graph' is too complex (9)
cosmos/converter.py:101:5: C901 'DbtToAirflowConverter.__init__' is too complex (9)
cosmos/dbt/parser/project.py:277:5: C901 'LegacyDbtProject.__post_init__' is too complex (10)
cosmos/dbt/selector.py:197:1: C901 'select_nodes' is too complex (9)

```
I picked out the functions which stood out to me as complex. If we want to reduce it to 6, there are plenty more functions to tackle. For some functions I would consider it unnecessary to drop it to 6. 
```shell
cosmos/config.py:108:9: C901 `__init__` is too complex (7 > 6)
cosmos/dbt/parser/project.py:63:9: C901 `_config_selector_ooo` is too complex (7 > 6)
cosmos/dbt/parser/project.py:98:5: C901 `extract_python_file_upstream_requirements` is too complex (7 > 6)
cosmos/dbt/parser/project.py:165:9: C901 `extract_sql_file_requirements` is too complex (7 > 6)
cosmos/dbt/selector.py:52:9: C901 `load_from_statement` is too complex (7 > 6)
cosmos/dbt/selector.py:119:9: C901 `_should_include_node` is too complex (7 > 6)
cosmos/hooks/subprocess.py:34:9: C901 `run_command` is too complex (8 > 6)
cosmos/operators/base.py:137:9: C901 `get_env` is too complex (7 > 6)
cosmos/operators/base.py:186:9: C901 `add_global_flags` is too complex (7 > 6)
cosmos/operators/local.py:136:9: C901 `store_compiled_sql` is too complex (7 > 6)
cosmos/profiles/base.py:183:9: C901 `get_dbt_value` is too complex (8 > 6)
Found 11 errors.
```

## Related Issue(s)
This PR continues on the amazing work of @jbandoro in on #525 
Related issue #641 

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -76,12 +76,6 @@ repos:
         name: mypy-python
         additional_dependencies: [types-PyYAML, types-attrs, attrs, types-requests, types-python-dateutil, apache-airflow]
         files: ^cosmos
-  - repo: https://github.com/pycqa/flake8
-    rev: 6.1.0
-    hooks:
-      - id: flake8
-        entry: pflake8
-        additional_dependencies: [pyproject-flake8]
```
[the remark this answers is not in the document]
Comment (tatiana) [reply .pre-commit-config.yaml]: Thanks, @joppevos ! That's great

File: cosmos/dbt/parser/project.py [hunk 2]
```diff
@@ -274,7 +274,7 @@ class LegacyDbtProject:
 
     operator_args: Dict[str, Any] = field(default_factory=dict)
 
-    def __post_init__(self) -> None:
+    def __post_init__(self) -> None:  # noqa: C901
```
Comment (tatiana) [inline cosmos/dbt/parser/project.py:277]: @joppevos Perhaps, as opposed to skipping the complexity check here, we could use the `or` strategy as well., e.g:
```
self.dbt_snapshots_dir = self.dbt_snapshots_dir or "snapshots"
```
What are your thoughts?

Comment (joppevos) [reply cosmos/dbt/parser/project.py]: Yes lets do that!

**Comment by tatiana** [comment]
@joppevos tiny error raised on the pre-commit check:

```
ruff..........................................................................Failed
- hook id: ruff
- exit code: 1

cosmos/converter.py:9:8: F811 Redefinition of unused `copy` from line 6
Found 1 error.
```

File: cosmos/converter.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import copy
 import inspect
 from typing import Any, Callable
+import copy
```
Comment (tatiana) [inline cosmos/converter.py:9]:
```suggestion
```

Comment (joppevos) [reply cosmos/converter.py]: Oeps - that slipped through in the solving the merge from console. thank you

**Comment by joppevos** [comment]
One more statement has gotten more complex than 8 after the merge to main. 
```
cosmos/dbt/selector.py:198:5: C901 'SelectorConfig.load_from_statement' is too complex (9)
```

I will try to solve it

**Review by tatiana — APPROVED** [review]
Thanks a lot for working on this, @joppevos ! It really helps with the overall code maintainability.

**Comment by joppevos** [comment]
> Thanks a lot for working on this, @joppevos ! It really helps with the overall code maintainability.
I'm happy to do it! Refactoring teaches me how Cosmos works on the inside.

**Pull request merged**